tests/50federation/00prepare.pl

```diff
@@ -279,14 +279,41 @@ sub send_and_await_outlier {
         Future->done(1);
       }),
     )->then( sub {
+      # the server may send a /state request; be prepared to answer that.
+      # (it may, alternatively, send individual /event requests)
+      my $state_req_fut = $inbound_server->await_request_v1_state(
+         $room_id, $outlier_event_id_Q,
+      )->then( sub {
+         my ( $req, @params ) = @_;
+         log_if_fail "/state request", \@params;
+
+         my $resp = {
+            pdus => [ values( %initial_room_state ) ],
+            auth_chain => [
+               map { $inbound_server->datastore->get_auth_chain_events( $room->id_for_event( $_ )) }
+                  values( %initial_room_state )
+               ],
+         };
+
+         log_if_fail "/state response", $resp;
+         $req->respond_json( $resp );
+
+         # return a future which never completes, so that wait_any is not
+         # satisfied.
+         return Future->new();
+      });
+
       # wait for S to turn up in /sync
```

Comment:
Out of interest, does the `/state` request only happen during the `/sync`, or can it happen before?

Reply:
hrm. It can happen before. Which probably means I've got this in the wrong place.

Reply:
(ftr: it's unclear if there's actually time for the `/state` request to arrive between us sending the response to `/state_ids` and creating the listener for `/state`, but it's probably best to move the listener setup earlier to be on the safe side)

Reply:
Right, I've now updated this. PTAL?